Template.go

```diff
@@ -183,6 +183,8 @@ func (t *Template) fileToXMLStruct(fname string) *xmlNode {
 func (t *Template) Params(v interface{}) {
 	// t.params = collectParams("", v)
 	switch val := v.(type) {
+	case map[string]interface{}:
```

Comment:
I did merged PR, but there was no issue (just no good comments).
There are some basic types, but complex types (like `map`) is sent through `StructParams`. Which now is confusing because there is similar func `StructToParams`.
`StructParams` will become `AnyToParams` to avoid confusion

```go
	default:
		if reflect.ValueOf(v).Kind() == reflect.Struct {
			t.params = StructToParams(val)
		} else {
			// any other type try to convert
			t.params = AnyToParams(val)
		}
	}
	```

It will be changed in next release. See: `t.params_test.go`